### ISSUE TITLE
Move overlord actions into worker. Have workers perform leader election.

### ISSUE THREAD

**Issue by ericchiang** [coreos/dex#302]
Discussed this with @sym3tri a while back. Would be a nice to have in the long term. Not a short term priority.

**Comment by ericchiang** [comment]
Bumping the priority due to #372. Practically, it's very hard to ensure that only one instance of the overlord will be running at a time in a clustered environment. This seems like a good solution to that.

Another issue to consider. Because the workers currently wait for connectors before starting, it's unclear how this would work without an overlord. Will have to address #14 before making progress on this.

**Comment by ericchiang** [comment]
The overlord is no more.
